canonical-web-and-design/assets.ubuntu.com
# add a tag table with many to many relation with asset & add search by tag name

## Done

- Add a `tag` table with a relation of type many to many with the table `asset`
- Add the ability to search by tag name alongside with the other existing search parameters (`q` and `type`)

## Ideas for improvement
- Add the possibility to search a list of tags
- Show the selected tags to search at the bottom of the search form, for example using [Chips](https://vanillaframework.io/docs/patterns/chip)

## QA

- Run the assets server locally with `dotrun`, note that there might be an issue with dotrun currently, you can run the project manually with the following commands:
```bash
docker-compose up -d
python3 -m venv .venv
pip3 install -r requirements.txt
pip3 install black python-dotenv flake8
# run the migrations
dotenv run alembic upgrade head
# start the server
dotenv run yarn start
```
- Run the command `update-assets-from-prod`: `FLASK_APP=webapp.app flask database update-assets-from-prod <in-prod-assets-api-token>`
- Go the the website: http://localhost:8017/manager
- Make sure that the assets are created successfully on your local server (you can search for the letter space in the search bar, as there are a lot of assets it will take a lot of time to load all the assets)
- Search for a simple query and then click on a tag to test the search by tag name

## Issues

Fixes #58 

## Screenshots
### The new searchable tags
![image](https://user-images.githubusercontent.com/36013798/140435200-b0e9babd-3891-44cc-bb7c-0ef9e915faad.png)
